### ISSUE TITLE
Cuda configure problem

### ISSUE THREAD

**Issue by nandra** [mesonbuild/meson#7643]
I'm trying to build simple app using yocto. I have a problem with configure phase which fails with:
```
add_languages_for
|     comp = self.environment.detect_compiler_for(lang, for_machine)
|   File "/home/marek/projects/test/build/tmp/work/armv8a_tegra194-poky-linux/test-app/git-r0/recipe-sysroot-native/usr/lib/python3.8/site-packages/mesonbuild/environment.py", line 1585, in detect_compiler_for
|     comp = self.compiler_from_language(lang, for_machine)
|   File "/home/marek/projects/test/build/tmp/work/armv8a_tegra194-poky-linux/test-app/git-r0/recipe-sysroot-native/usr/lib/python3.8/site-packages/mesonbuild/environment.py", line 1563, in compiler_from_language
|     comp = self.detect_cuda_compiler(for_machine)
|   File "/home/marek/projects/test/build/tmp/work/armv8a_tegra194-poky-linux/test-app/git-r0/recipe-sysroot-native/usr/lib/python3.8/site-packages/mesonbuild/environment.py", line 1129, in detect_cuda_compiler
|     linker = CudaLinker(compiler, for_machine, 'nvlink', CudaCompiler.LINKER_PREFIX, [], version=CudaLinker.parse_version())
|   File "/home/marek/projects/test/build/tmp/work/armv8a_tegra194-poky-linux/test-app/git-r0/recipe-sysroot-native/usr/lib/python3.8/site-packages/mesonbuild/linkers.py", line 963, in __init__
|     super().__init__('nvlink', *args, **kwargs)
| TypeError: __init__() takes 6 positional arguments but 7 positional arguments (and 1 keyword-only argument) were given
| The Meson build system
| Version: 0.53.2
``` 

- cross build fox xavier NX
- Ubuntu 18.04 is host machine and building in Yocto build system
- python 3.8
- Meson 0.53.2

**Comment by mensinda** [comment]
Could you please test if this problem also occurs with meson 0.55.1?

**Comment by nandra** [comment]
Looks like `0.53.1` and `0.55.1` don't have this issue.

**Comment by mensinda** [comment]
Closing, since this is now fixed and there won't be any updates for the 0.53 branch.
